monitoring/monitorlib/auth.py

```diff
@@ -254,29 +253,46 @@ def issue_token(self, intended_audience: str, scopes: List[str]) -> str:
 
 
 class ClientIdClientSecret(AuthAdapter):
-  """Auth adapter that gets JWTs using a client ID and client secret."""
+  """Auth adapter that gets JWTs using a client ID and client secret. By default, this will send the request as JSON, you can use send_request_as_data flag to send the request as form data. """
 
-  def __init__(self, token_endpoint: str, client_id: str, client_secret: str):
+  def __init__(self, token_endpoint: str, client_id: str, client_secret: str, send_request_as_data :bool=False):
     super().__init__()
 
     self._oauth_token_endpoint = token_endpoint
     self._client_id = client_id
     self._client_secret = client_secret
-
+    self.send_request_as_data = send_request_as_data
+    
   # Overrides method in AuthAdapter
   def issue_token(self, intended_audience: str, scopes: List[str]) -> str:
-    response = requests.post(self._oauth_token_endpoint, json={
+    payload = {
       'grant_type': 'client_credentials',
       'client_id': self._client_id,
       'client_secret': self._client_secret,
       'audience': intended_audience,
       'scope': ' '.join(scopes),
-    })
+    }
+
+    if self.send_request_as_data: 
+      response = requests.post(self._oauth_token_endpoint, data=payload)
+    else:
+      response = requests.post(self._oauth_token_endpoint, json=payload)
     if response.status_code != 200:
       raise AccessTokenError('Unable to retrieve access token:\n' + response.content.decode('utf-8'))
     return response.json()['access_token']
 
 
+class FlightPassport(ClientIdClientSecret, AuthAdapter):
```

Comment:
`ClientIdClientSecret` is already an `AuthAdapter` (no need for multiple inheritance), so I think we want:

```python
class FlightPassport(ClientIdClientSecret):
  def __init__(self, token_endpoint: str, client_id: str, client_secret: str):
    super(FlightPassport, self).__init__(token_endpoint, client_id, client_secret, True)
```

(but be careful of the value of that last argument if my other comment is addressed; `True` is not a `str`)

Reply:
Thanks, yes I have now put in a check and converts it into a bool